File: 준수/PGS_42577.java
import java.util.*;

class Solution {
    
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        ArrayList<String> arr = new ArrayList<String>();
        
        Arrays.sort(phone_book);
        
        for (int i = 0; i < phone_book.length - 1; i++) {
            if (!answer) break;
            String target = phone_book[i];
            
            for (int k = 0; k < target.length(); k++) {
                if (target.charAt(k) != phone_book[i + 1].charAt(k)) break;

                if (k == target.length() - 1) answer = false;
            }
        }
        
        return answer;
    }
}

File: 유나/PGS 42577.py
def solution(phone_book):
    answer = True
    # 정렬
    phone_book.sort()
    for idx in range(len(phone_book)-1):
        # 해당 번호가 다음 번호에 접두사인지 확인
        if phone_book[idx] == phone_book[idx+1][:len(phone_book[idx])]:
            answer = False
    return answer
